src/renderer/html_handlebars/search.rs

```diff
@@ -0,0 +1,221 @@
+extern crate elasticlunr;
+
+use std::borrow::Cow;
+use std::collections::HashMap;
+use std::path::Path;
+
+use pulldown_cmark::*;
+use serde_json;
+use self::elasticlunr::Index;
+
+use book::BookItem;
+use config::Search;
+use errors::*;
+use renderer::RenderContext;
+use utils;
+use theme::searcher::Searcher;
+
+/// Creates all files required for search.
+pub fn create_files(ctx: &RenderContext) -> Result<()> {
+    let search_config = ctx.config.html_config().unwrap_or_default().search;
```

Comment:
Is there any chance we can pass in a `&HtmlConfig` here? It seems such a waste to be re-deserializing it when the caller should already have one lying around.